udata/core/followers/api.py

```diff
@@ -46,7 +46,12 @@ def get(self, id):
     @api.doc(id='follow', description=NOTE)
     def post(self, id):
         '''Follow an object given its ID'''
-        follow, created = self.model.objects.get_or_create(follower=current_user.id, following=id, until=None)
+        try:
+            follow = self.model.objects.get(follower=current_user.id, following=id, until=None)
+            created = False
+        except self.model.DoesNotExist:
+            follow = self.model.objects.create(follower=current_user.id, following=id, until=None)
+            created = True
```

Comment:
I'm a bit afraid of adding a method they removed upstream because of the [impossibility to do that reliably with Mongo](https://github.com/MongoEngine/mongoengine/issues/300#issuecomment-111854695).

We might want to document why we think that we can do that safely for each of our case though.